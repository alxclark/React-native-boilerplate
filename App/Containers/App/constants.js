export const INCREMENT_COUNTER = "[App] INCREMENT_COUNTER";
