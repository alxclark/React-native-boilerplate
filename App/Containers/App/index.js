import React, { Component } from "react";
import { StyleSheet, Text, View } from "react-native";
import { Provider } from "react-redux";

import createStore from "../../Redux/index";
import DebugConfig from "../../Config/DebugConfig";

const store = createStore();

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center"
  }
});

class App extends Component {
  render() {
    console.tron.log(store.getState());
    return (
      <Provider store={store}>
        <View style={styles.container}>
          <Text>Open up App.js to start working on your app!</Text>
          <Text>Test allo</Text>
          <Text>Changes you make will automatically reload.</Text>
          <Text>Shake your phone to open the developer menu.</Text>
        </View>
      </Provider>
    );
  }
}

// allow reactotron overlay for fast design in dev mode
export default (DebugConfig.useReactotron ? console.tron.overlay(App) : App);
//export default App;
