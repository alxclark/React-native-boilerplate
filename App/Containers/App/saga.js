export function* appSaga(action) {
  console.tron.log("It worked");
}
