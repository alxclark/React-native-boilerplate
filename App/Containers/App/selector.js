import { createSelector } from "reselect";

const selectGlobal = state => state.get("global");

const makeSelectIndex = () =>
  createSelector(selectGlobal, globalState => globalState.get("index"));

export { makeSelectIndex };
