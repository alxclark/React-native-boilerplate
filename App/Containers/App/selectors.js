import { createSelector } from "reselect";

const selectGlobal = state => {
  return state.get("global");
};

const makeSelectCounter = () =>
  createSelector(selectGlobal, globalState => globalState.get("counter"));

export { selectGlobal, makeSelectCounter };
