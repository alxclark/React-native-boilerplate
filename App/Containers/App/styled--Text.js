import styled from "styled-components";

const StyledText = styled.Text`
  color: black;
`;

export default StyledText;
