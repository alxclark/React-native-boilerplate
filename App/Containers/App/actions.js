import { ADD_TODO } from "./constants";

export function addTodo() {
  return {
    type: ADD_TODO
  };
}
