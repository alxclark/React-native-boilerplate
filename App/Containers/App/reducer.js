import { fromJS } from "immutable";

import { ADD_TODO } from "./constants";

// The initial state of the App
const initialState = fromJS({
  index: 2
});

export default function AppReducer(state = initialState, action) {
  switch (action.type) {
    case ADD_TODO:
      console.tron.log("YES");
      return state;
    default:
      return state;
  }
}
