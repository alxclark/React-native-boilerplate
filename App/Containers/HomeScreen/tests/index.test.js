import "react-native";
import React from "react";

import renderer from "react-test-renderer";

import { HomeScreen, mapDispatchToProps } from "../index";
import { incrementCounter } from "../../App/actions";

describe("<HomeScreen />", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(
        <HomeScreen
          counter={0}
          onIncrementCounter={() => console.log("test function")}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("mapDispatchToProps", () => {
    describe("onIncrementCounter", () => {
      it("should be injected", () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onIncrementCounter).toBeDefined();
      });

      it("should dispatch incrementCounter when called", () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.onIncrementCounter();
        expect(dispatch).toHaveBeenCalledWith(incrementCounter());
      });
    });
  });
});
