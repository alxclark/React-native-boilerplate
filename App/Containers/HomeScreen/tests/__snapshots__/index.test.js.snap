// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "lightblue",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "black",
        },
        undefined,
      ]
    }
  >
    React native boilerplate
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        0
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "black",
        },
        undefined,
      ]
    }
  >
    Press on button to increment the counter
  </Text>
</View>
`;
